analysis/study_definition.py

```diff
@@ -1,168 +1,376 @@
+    population=patients.all(),
+
     age=patients.age_as_of(
-        "2020-02-01",
+        "2021-01-01",
         return_expectations={
-            "rate": "universal",
-            "int": {"distribution": "population_ages"},
+            "int": {'distribution': 'normal', 'mean': 40, 'stddev': 20},
         },
     ),
-    
-    # https://github.com/opensafely/risk-factors-research/issues/46
+
     sex=patients.sex(
         return_expectations={
-            "rate": "universal",
-            "category": {"ratios": {"M": 0.49, "F": 0.51}},
-        }
+            "category": {'ratios': {'F': 0.495, 'M': 0.495, '': 0.01}},
+        },
     ),
-    
-    # https://codelists.opensafely.org/codelist/opensafely/chronic-cardiac-disease/2020-04-08/
-    chronic_cardiac_disease=patients.with_these_clinical_events(
-        chronic_cardiac_disease_codes,
+
+    # Asthma Diagnosis code
+    ast=patients.with_these_clinical_events(
+        codelists.ast,
         returning="date",
-        find_first_match_in_period=True,
-        include_month=True,
-        return_expectations={"incidence": 0.2},
+        date_format="YYYY",
+        find_last_match_in_period=True,
     ),
 
-    # https://codelists.opensafely.org/codelist/opensafely/chronic-liver-disease/2020-06-02/
-    chronic_liver_disease=patients.with_these_clinical_events(
-        chronic_liver_disease_codes,
+    # Asthma Admission codes
+    astadm=patients.with_these_clinical_events(
+        codelists.astadm,
         returning="date",
-        find_first_match_in_period=True,
-        include_month=True,
-        return_expectations={
-            "incidence": 0.2,
-            "date": {"earliest": "1950-01-01", "latest": "today"},
-        },
+        date_format="YYYY",
+        find_last_match_in_period=True,
     ),
 
-    # https://github.com/opensafely/risk-factors-research/issues/51
-    bmi=patients.most_recent_bmi(
-        on_or_after="2010-02-01",
-        minimum_age_at_measurement=16,
-        include_measurement_date=True,
-        include_month=True,
-        return_expectations={
-            "incidence": 0.6,
-            "float": {"distribution": "normal", "mean": 35, "stddev": 10},
-        },
+    # Asthma systemic steroid prescription codes
+    astrx=patients.with_these_medications(
+        codelists.astrx,
+        returning="date",
+        date_format="YYYY",
+        find_last_match_in_period=True,
     ),
 
-    # https://github.com/opensafely/risk-factors-research/issues/48
-    bp_sys=patients.mean_recorded_value(
-        systolic_blood_pressure_codes,
-        on_most_recent_day_of_measurement=True,
-        on_or_before="2020-02-01",
-        include_measurement_date=True,
-        include_month=True,
-        return_expectations={
-            "incidence": 0.6,
-            "float": {"distribution": "normal", "mean": 80, "stddev": 10},
-        },
+    # BMI
+    bmi=patients.with_these_clinical_events(
```

Comment:
for BMI do we need to also return categories for comparison, rather than only comparing prevalence of "having BMI recorded"?

Reply:
Probably a good idea, but could wait until we have the full study definition, so that we can compare the actual groups used in the study.   What do you think?

Reply:
OK great!